tools/make-autosuspend-rules.py

```diff
@@ -2,32 +2,23 @@
 # SPDX-License-Identifier: LGPL-2.1+
 
 # Generate autosuspend rules for devices that have been whitelisted (IE tested)
-# by the Chromium OS team. Please keep this script in sync with:
+# by the Chromium OS team. Based on
 # https://chromium.googlesource.com/chromiumos/platform2/+/master/power_manager/udev/gen_autosuspend_rules.py
 
-import sys
 import chromiumos.gen_autosuspend_rules
 
-HWDB_FILE = """\
-%(usb_entries)s\
-%(pci_entries)s\
-"""
-
-if __name__ == '__main__':
-    if len(sys.argv) > 1:
-        sys.stdout = open(sys.argv[1], 'w')
-
-    pci_entries = ''
-    for dev_ids in chromiumos.gen_autosuspend_rules.PCI_IDS:
-        vendor, device = dev_ids.split(':')
-
-        pci_entries += ('usb:v%sp%s*\n'
-                        ' ID_AUTOSUSPEND=1\n' % (vendor, device))
-    usb_entries = ''
-    for dev_ids in chromiumos.gen_autosuspend_rules.USB_IDS:
-        vendor, device = dev_ids.split(':')
-
-        usb_entries += ('pci:v%sp%s*\n'
-                        ' ID_AUTOSUSPEND=1\n' % (vendor, device))
-
-    print(HWDB_FILE % {'pci_entries' : pci_entries, 'usb_entries': usb_entries})
+print('# <subsystem>:v<VENDOR>d<DEVICE>')
+for entry in chromiumos.gen_autosuspend_rules.PCI_IDS:
+    vendor, device = entry.split(':')
+    vendor = int(vendor, 16)
+    device = int(device, 16)
+    print(f'pci:v{vendor:08X}d{device:08X}*')
+
+print('# <subsystem>:v<VENDOR>p<PRODUCT>')
+for entry in chromiumos.gen_autosuspend_rules.USB_IDS:
+    vendor, product = entry.split(':')
+    vendor = int(vendor, 16)
+    product = int(product, 16)
+    print(f'usb:v{vendor:04X}p{product:04X}*')
+
+print(' AUTOSUSPEND=1')
```

Comment:
actually, historicaly the ID prefix used to just mean data generated from one of the xyz_id tools. But it then morphed into a namespace thing to differentiate between kernel props and userspace props. And that's a big issue still, since the kernel's device props are generally not prefixed and we somehow need to avoid clashing with those names.

for the ACCEL and KEYBOARD stuff we should have used the ID_ prefix too, it probably was an accident that it wasn't, but the maintainers of the relevant subsystems in the kernel are sympathetic to our cause, so i wouldn't expect name clashes there. However, the AUTOSUSPEND applied across subsystems is much riskier business, as the kernel maintainers are probably not as sympathetic to us and might just add a property named like that to their props too. hence here too I'd just stick to the "ID_" prefix to make clear it's usperspace stuff, not kernelspace stuff.